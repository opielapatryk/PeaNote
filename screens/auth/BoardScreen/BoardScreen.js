import React,{ useEffect } from 'react';
import { Pressable,View,FlatList } from 'react-native';
import { renderNotes } from './apiBoardScreen';
import { useDispatch, useSelector } from 'react-redux';
import { styles } from '../../../assets/styles/styles';
import { fetchNotes, checkThenChangeInfo } from './apiBoardScreen';
import { useFocusEffect } from '@react-navigation/native';

const BoardScreen = () => {
  const { notes } = useSelector((state) => state.board);
  const dispatch = useDispatch()

  useEffect(()=>{
    fetchNotes(dispatch);
  },[])

  useFocusEffect(
    React.useCallback(() => {
      fetchNotes(dispatch);
    }, [])
  );

  return (
    <View style={styles.flexone}>
      <Pressable onPress={() => checkThenChangeInfo(dispatch,notes)} style={styles.board}>
        <FlatList data={notes} renderItem={renderNotes} keyExtractor={note => note.id}/>
      </Pressable>
    </View>
  );
};

export default BoardScreen;