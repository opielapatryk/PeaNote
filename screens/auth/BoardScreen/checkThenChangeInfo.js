import {changeInfo} from '../../../store/notes/boardSlice';

export const checkThenChangeInfo = (dispatch, notes) => {
    notes.map((note) => {
      if(note.isInfo === true)
      {
        dispatch(changeInfo(note.id))
      }
    })
  }