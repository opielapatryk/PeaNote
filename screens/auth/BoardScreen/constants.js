import auth from '@react-native-firebase/auth';

export const MY_EMAIL = 'patryk.opiela02@gmail.com'