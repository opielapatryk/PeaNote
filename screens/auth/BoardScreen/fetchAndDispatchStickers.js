export const fetchAndDispatchStickers = async (stickers, dispatch,addNoteAction) => {
    if(stickers){
      stickers.forEach((sticker,index) => dispatch(addNoteAction({ id: index + 1, text: sticker.content, isInfo: false })));
    }
  };