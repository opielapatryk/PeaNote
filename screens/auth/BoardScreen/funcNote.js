import {changeInfo,removeNote} from '../../../store/notes/boardSlice'
import auth, { firebase } from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';

let numberOfDeleted = 0

export const handlePress = (notes,dispatch,isInfo,id,animatedValues) => {

    {notes.forEach(note => {
        if(note.isInfo === true){
        dispatch(changeInfo(note.id));
        }
    });}
    if (isInfo) {
        dispatch(changeInfo(id));
        dispatch(removeNote(id));
        deleteNote(id);
    } else {
        dispatch(changeInfo(id));
    }
};

const deleteNote = async (id) => {

      
    const MY_EMAIL = auth().currentUser.email

    try {

        
        // TAKE STICKERS ON BOARD
        let stickersonboard

        const result = await firestore()
        .collection('users')
        .where('email', '==', MY_EMAIL)
        .get()
    
        result.forEach(doc=>{
            stickersonboard = doc.data().stickersOnBoard
        })
  
        // ITERATE OVER STICKERS ON BOARD
  
        stickersonboard.forEach((sticker,index) => {
          index = index + 1
          let sum = id - numberOfDeleted
          if(index === sum){
            creator = sticker.creator
            content = sticker.content
          }
        })
        numberOfDeleted++

         // REMOVE STICKER FROM FIRESTORE 
        firestore()
        .collection('users')
        .where('email', '==', MY_EMAIL)
        .get()
        .then((querySnapshot)=>{
        querySnapshot.forEach(doc => {
            firestore()
            .collection('users')
            .doc(doc.id)
            .update({
            stickersOnBoard: firebase.firestore.FieldValue.arrayRemove({
                content: content,
                creator: creator,
            }),
            })
        })
        })

        // MANAGE REDUX STORE
        
    } catch (error) {
      console.log(error.message);
    }
}


