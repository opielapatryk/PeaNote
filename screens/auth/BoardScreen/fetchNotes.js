import {addNote,addPendingNote} from '../../../store/notes/boardSlice';
import firestore from '@react-native-firebase/firestore';
import { fetchAndDispatchStickers } from './fetchAndDispatchStickers'
import { MY_EMAIL } from '../../constants';

export const fetchNotes = async (dispatch) => {  
    try {
      let stickersonboard 
      let pending
      
      const result = await firestore()
      .collection('users')
      .where('email', '==', MY_EMAIL)
      .get()
  
      result.forEach(doc=>{
        pending = doc.data().pending
        stickersonboard = doc.data().stickersOnBoard
      })
      
      await fetchAndDispatchStickers(stickersonboard, dispatch, addNote);
      await fetchAndDispatchStickers(pending, dispatch, addPendingNote);
    } catch (error) {
      console.log(error);
    }
  };