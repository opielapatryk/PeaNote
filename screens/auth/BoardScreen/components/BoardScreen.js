import React,{ useEffect } from 'react';
import { Pressable,View,FlatList } from 'react-native';
import { renderNotes } from '../functions/renderNotes';
import { useDispatch, useSelector } from 'react-redux';
import { styles } from '../../../../assets/styles/styles';
import { checkThenChangeInfo } from '../functions/checkThenChangeInfo';
import { useFocusEffect } from '@react-navigation/native';
import { fetchNotes } from '../functions/fetchNotes';
import { KEY_EXTRACTOR_NOTES } from '../../../constants';

// Here are displayed all approved notes
const BoardScreen = () => {
  const { notes } = useSelector((state) => state.board);
  const dispatch = useDispatch()

  useEffect(()=>{
    fetchNotes(dispatch);
  },[])

  useFocusEffect(
    React.useCallback(() => {
      fetchNotes(dispatch);
    }, [])
  );

  return (
    <View style={styles.flexone}>
      <Pressable onPress={() => checkThenChangeInfo(dispatch,notes)} style={styles.board}>
        <FlatList numColumns={2} data={notes} renderItem={renderNotes} keyExtractor={KEY_EXTRACTOR_NOTES}/>
      </Pressable>
    </View>
  );
};

export default BoardScreen;