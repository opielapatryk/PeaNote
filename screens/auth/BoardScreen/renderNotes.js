import {Note} from '../../../components/Note'

export const renderNotes = ({item}) => {
    return (
      <Note id={item.id} isInfo={item.isInfo} />
    )
  }