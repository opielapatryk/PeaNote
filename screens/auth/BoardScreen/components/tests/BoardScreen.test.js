import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';  
import store from '../../../../../store/store'; 
import BoardScreen from '../BoardScreen';
import { NavigationContainer } from '@react-navigation/native';
import { act } from 'react-test-renderer';

jest.mock('@react-native-firebase/auth', () => () => {
    return {
        currentUser: {
            email:'test@example.com'
        }
    }
  })

jest.mock('@react-native-firebase/firestore', () => ({
    collection: jest.fn(),
    doc: jest.fn(),
    get: jest.fn(),
}));


test('board screen renders correctly', async () => {
  const tree = renderer.create(
    <Provider store={store}>
        <NavigationContainer>
            <BoardScreen />
        </NavigationContainer>
    </Provider>
  ).toJSON();

  await act(async () => {
    expect(tree).toMatchSnapshot();
  })
});
