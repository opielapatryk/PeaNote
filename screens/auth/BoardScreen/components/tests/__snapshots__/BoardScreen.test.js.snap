// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board screen renders correctly. 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFDF3",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
